src/blk/kernel/KernelDevice.cc

```diff
@@ -82,17 +82,31 @@ KernelDevice::KernelDevice(CephContext* cct, aio_callback_t cb, void *cbpriv, ai
   }
 }
 
+inline int KernelDevice::choose_fd(bool buffered, int write_hint) const
```

Comment:
@wjwithagen you have at least three changes in this single commit. none of them is explained in the commit message.

1. relocate`KernelDevice::choose_fd()`
1. define `BUFFEREDFD` and `DIRECTFD`. could use a scoped enum.
1. replace more occurrences of `fd_directs[]` with `choose_fd()` .

could you please split this commit into smaller ones and accompany them with commit message explaining why and what?

Reply:
Oke, I will, But:
ad1) It was just in the code at a place where it was declared after usage...
ad2) I just tried, but that explodes all the way up into `BlueStore.cc` and `BlueFS.cc`, so that is going to be a big patch.
ad3) That was more or less why I got into this PR.

Reply:
regarding 2), if that's the case i'd keep as it is. to define `BUFFEREDFD` and `DIRECTFD` as booleans is but confusing.

Reply:
> 
> 
> > Oke, I will, But:
> > ad1) It was just in the code at a place where it was declared after usage...
> 
> i don't get why this change is necessary. `KernelDevice::choose_fd()` is already declared in the header file.

It was just thrown somewhere no logical in the code (by me, I think) some time ago, when Sage introduced direct and buffered file access. Which breaks real hard on FreeBSD.
So yes, It could be considered cosmetic, but I always prefer to put definition before usage.

> > ad2) I just tried, but that explodes all the way up into BlueStore.cc and BlueFS.cc, so that is going to be a big patch.
> 
> regarding 2), if that's the case i'd keep as it is. to define `BUFFEREDFD` and `DIRECTFD` as booleans is but confusing.

Well actually the change is not bad because it removes quite a few true and fals at odd places with consts that are much understandable.... So it is an improvement for the code.
Maybe I'll just push it, and you can see it for yourself. 
Still need to add 2 helper routines to fix printing, and `bool` to `FAMode`  (FileAccesMode)